### PR TITLE
feat: arrow improvements

### PR DESCRIPTION
**Pull request by Manukyanq**
https://github.com/Clariity/react-chessboard/assets/42891870/2abc14af-54e5-418b-8adb-943078cf37dc

### CODE REVIEW

**Comment by Clariity** [comment]
This looks amazing!

My only request is can the arrows finish just before the centre of the square they point to, rather than past the centre / near the end of the square?

This way multiple arrows can point towards the same square without massively overlapping

**Comment by Manukyanq** [comment]
> My only request is can the arrows finish just before the centre of the square they point to, rather than past the centre / near the end of the square?

Hi, please check demo [here](https://react-chessboard-git-fork-manukyanq-feature-arr-14102b-clariity.vercel.app/?path=/story/example-chessboard--configurable-board), I've made them shorter already, screen recording in the video demo is not actual now

**Comment by Clariity** [comment]
Amazing

2 slight changes to make this perfect:

1. can the arrow head be moved ever so slightly further into the square as it looks a bit weird with the base of the arrow head being directly in line with the square edge itself
2. There is a very slight gap between the the arrow head and the arrow body that needs fixing, maybe only 1 or 2 pixels

File: src/chessboard/components/Arrows.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        // if there are differnet arrows targeting same square make thier length a bit shorter
+        if (
+          arrows.some(
+            (restArrow) =>
+              restArrow[0] !== arrow[0] && restArrow[1] === arrow[1]
+          ) &&
+          !isArrowActive
+        ) {
+          ARROW_LENGTH_REDUCER = boardWidth / 16;
+        }
```
Comment (Clariity) [inline src/chessboard/components/Arrows.tsx:45]: Can I see what it looks like without making their lengths shorter when targeting the same square, this functionality seemed a bit off, so I'm curious to see what it's like as normal and then if I agree it needs shortening we can add this back

Comment (Manukyanq) [reply src/chessboard/components/Arrows.tsx:45]: https://github.com/Clariity/react-chessboard/assets/42891870/c6f4d7d6-5627-432e-a30e-b724b65563f8

Comment (Clariity) [reply src/chessboard/components/Arrows.tsx:45]: I think you've made a great decision here then with the shortening on the same square, I'll take some time soon to give the code a good review but I think this should be merged in soon, great job

Comment (Manukyanq) [reply src/chessboard/components/Arrows.tsx:45]: Thanks

**Review by Clariity — APPROVED** [review]
fantastic stuff

**Pull request merged**